src/FrequencyMapAgent.cpp

```diff
@@ -410,13 +454,31 @@ namespace geopm
             return;
         }
         m_platform_io.write_control("CPU_FREQUENCY_MAX_CONTROL", GEOPM_DOMAIN_BOARD, 0,
-                                    policy[M_POLICY_FREQ_DEFAULT]);
+                                    policy[M_POLICY_FREQ_CPU_DEFAULT]);
         if (!std::isnan(policy[M_POLICY_FREQ_UNCORE])) {
                 m_platform_io.write_control("CPU_UNCORE_FREQUENCY_MIN_CONTROL", GEOPM_DOMAIN_BOARD, 0,
                                             policy[M_POLICY_FREQ_UNCORE]);
                 m_platform_io.write_control("CPU_UNCORE_FREQUENCY_MAX_CONTROL", GEOPM_DOMAIN_BOARD, 0,
                                             policy[M_POLICY_FREQ_UNCORE]);
         }
+
+        //Apply GPU default frequency settings
+        //Ensuring that MIN_CONTROL <= MAX CONTROL at all times to avoid issues
+        if (!std::isnan(policy[M_POLICY_FREQ_GPU_DEFAULT]) && m_do_gpu_ctl) {
+            double gpu_min_control = m_platform_io.read_signal("GPU_CORE_FREQUENCY_MIN_CONTROL", m_gpu_freq_ctl_domain_type, 0);
+            if (policy[M_POLICY_FREQ_GPU_DEFAULT] >= gpu_min_control) {
+                m_platform_io.write_control("GPU_CORE_FREQUENCY_MAX_CONTROL", GEOPM_DOMAIN_BOARD, 0,
+                                            policy[M_POLICY_FREQ_GPU_DEFAULT]);
+                m_platform_io.write_control("GPU_CORE_FREQUENCY_MIN_CONTROL", GEOPM_DOMAIN_BOARD, 0,
+                                            policy[M_POLICY_FREQ_GPU_DEFAULT]);
+            }
+            else  {
+                m_platform_io.write_control("GPU_CORE_FREQUENCY_MIN_CONTROL", GEOPM_DOMAIN_BOARD, 0,
+                                            policy[M_POLICY_FREQ_GPU_DEFAULT]);
+                m_platform_io.write_control("GPU_CORE_FREQUENCY_MAX_CONTROL", GEOPM_DOMAIN_BOARD, 0,
+                                            policy[M_POLICY_FREQ_GPU_DEFAULT]);
+            }
+        }
```

Comment:
Something is off: I'm seeing that regardless of the branch that's taken, the side effect is going to be the same: to set GPU min and max to the policy default for GPU.

Reply:
The ordering is important on some architectures that enforce that min < max. If the new setting is less than the system's current min, we must set the min first, otherwise the max can get overwritten by a non-geopm mechanism. If the new setting is greater than the current min, it is safe to set the max first.